source/core/shape.cc

```diff
@@ -104,6 +125,140 @@ Shape<dim>::align_and_center(const Point<dim> &evaluation_point) const
   return translated_point;
 }
 
+template <int dim>
+Point<dim>
+Shape<dim>::reverse_align_and_center(const Point<dim> &evaluation_point) const
+{
+  // Translation and rotation to standard position and orientation for
+  // distance calculations
+  Point<dim> center_of_rotation = position;
+
+  Point<dim> rotated_point;
+  Point<dim> translated_point;
+
+  // Rotation from the solid orientation
+  // Angular position around x, y and z axis
+  Tensor<1, 3> theta = orientation;
```

Comment:
I'd look over the newlines in this file, so make more coherent 'code' paragraphs